gcloud/pubsub/topic.py

```diff
@@ -240,24 +234,14 @@ def list_subscriptions(self, page_size=None, page_token=None, client=None):
                   value as ``page_token``).
         """
         client = self._require_client(client)
-        params = {}
-
-        if page_size is not None:
-            params['pageSize'] = page_size
-
-        if page_token is not None:
-            params['pageToken'] = page_token
-
-        path = '/projects/%s/topics/%s/subscriptions' % (
-            self.project, self.name)
-
-        resp = client.connection.api_request(method='GET', path=path,
-                                             query_params=params)
+        conn = client.connection
+        sub_paths, next_token = conn.topic_list_subscriptions(
+            self.full_name, page_size, page_token)
         subscriptions = []
-        for sub_path in resp.get('subscriptions', ()):
+        for sub_path in sub_paths:
```

Comment:
The topic-specific API doesn't return reprs :().  `sub_paths` is a list of strings.